### DISCUSSION TITLE
Elasticsearch Extra hosts error

### DISCUSSION THREAD

**Discussion by wolfmatt21** [Security-Onion-Solutions/securityonion#5765] [Unsupported Versions] [answered]
HI, 

My manager node is having issues with Elasticsearch and when i have gone to force stop it to start it again it shows up this issue:

![image](https://user-images.githubusercontent.com/82594848/136008322-0be62ee2-cd63-4309-b8e9-175012ef692f.png)

Looks like someone else had this issue in May but there was no answer. Has anyone else experienced this?

Im running .80 version but was happening with SO .70 aswell

**Comment by dougburks** [comment]
Yeah, we did have the report in May, but I don't think we've ever been able to duplicate this issue. Can you provide more details about your deployment and any customizations you have made?

**Reply by wolfmatt21** [reply]
Hi Doug,

Ye i think it has something to do with adding of a sensor which has gone a but wrong and salt is having difficulty with it. Ive looked through the salt file local and default and cant see anything on the master(opt/so/salt...).

Its a 7 Sensor deployment( 1 manager/search, then 6 heavynodes). Salt is picking up all the sensors fine and so-status is green for all nodes but manager.(see below). When i force stop elasticsearch then the extra.hosts issue happens. If i cant get this working by tomorrow I will have to run a resetup of the entiresud  deployment which i really dont want to do.

Global.sls conf

global:
  soversion: 2.3.80
  hnmanager: '10.0.0.0/8,192.168.0.0/16,172.16.0.0/12'
  dockernet: '172.17.0.0'
  mdengine: 'ZEEK'
  ids: 'Suricata'
  url_base: '10.***'
  managerip: '10.***'
  airgap: False
  hivekey: 'ap0vajJvJKLnsjWXD4rs'
  hiveplaysecret: '***'
  cortexkey: '***'
  cortexorgname: 'SecurityOnion'
  cortexorguserkey: '***'
  cortexplaysecret: '**'
  fleet_custom_hostname:
  fleet_manager: True
  fleet_node: False
  fleet_packages-timestamp: '2021-09-12-14:16'
  fleet_packages-version: 4
  fleet_hostname: '***'
  fleet_ip: '***'
  sensoronikey: '***'
  wazuh: 0
  imagerepo: 'security-onion-solutions'
  pipeline: 'redis'
sensoroni:
  node_checkin_interval_ms: 10000
strelka:
  enabled: 1
  rules: 1
  repos:
    - 'https://github.com/Neo23x0/signature-base'
curator:
  hot_warm: False
elastic:
  features: False
elasticsearch:
  replicas: 0
  true_cluster: False
  true_cluster_name: 'so'
  discovery_nodes: 1
  hot_warm_enabled: False
  index_settings:
    so-redis:
      shards: 1
      warm: 7
      close: 30
      delete: 365
    so-kibana:
      shards: 1
      warm: 7
      close: 30
      delete: 365
    so-logstash:
      shards: 1
      warm: 7
      close: 30
      delete: 365
    so-elasticsearch:
      shards: 1
      warm: 7
      close: 30
      delete: 365
    so-beats:
      shards: 1
      warm: 7
      close: 30
      delete: 365
    so-firewall:
      shards: 1
      warm: 7
      close: 30
      delete: 365
    so-flow:
      shards: 1
      warm: 7
      close: 30
      delete: 365
    so-ids:
      shards: 1
      warm: 7
      close: 30
      delete: 365
    so-import:
      shards: 1
      warm: 7
      close: 73000
      delete: 73001
    so-osquery:
      shards: 1
      warm: 7
      close: 30
      delete: 365
    so-ossec:
      shards: 1
      warm: 7
      close: 30
      delete: 365
    so-strelka:
      shards: 1
      warm: 7
      close: 30
      delete: 365
    so-syslog:
      shards: 1
      warm: 7
      close: 30
      delete: 365
    so-zeek:
      shards: 5
      warm: 7
      close: 45
      delete: 365
minio:
  access_key: ''
  access_secret: ''
s3_settings:
  size_file: 2048
  time_file: 1
  upload_queue_size: 4
  encoding: 'gzip'
  interval: 5
backup:
  locations:
    - /opt/so/saltstack/local
soctopus:
  playbook:
    rulesets:
      - windows
docker:
  range: '172.17.0.0/24'
  bip: '172.17.0.1/24'
redis_settings:
  redis_maxmemory: 6000
playbook:
  api_key: ae9e436b808e983163002ef6a43e5500c3247bd8
![image](https://user-images.githubusercontent.com/82594848/136199884-c8d1a9a1-4b8f-424d-bc1f-6e8000f2a454.png)


![image](https://user-images.githubusercontent.com/82594848/136199691-e22f5da7-75f7-42e7-a80e-acb59732fd07.png)




see below for information-

![image](https://user-images.githubusercontent.com/82594848/136199279-02e41cc8-289a-4cec-acc6-9c45689c39d2.png)

**Reply by wolfmatt21** [reply]
Just adding more infomation when i use the sttart elastic or stop elastic force it brings that message up on this process-

![image](https://user-images.githubusercontent.com/82594848/136202206-e5edbc98-8db7-4042-88bf-81af1c699ddf.png)


Where is the conf for this script kept?

**Comment by m0duspwnens** [comment]
@wolfmatt21 It appears that `2` has been inserted as an IP address somewhere.  Can you check by running the following commands on the manager?

`salt-call pillar.get elasticsearch:mainip`
`salt-call pillar.get global:managerip`
`salt-call pillar.get nodestab`

Looking at the global.sls, not sure if this has been redacted or not, but the IPs for `url_base` and `managerip` incomplete.
```
global:
soversion: 2.3.80
hnmanager: '10.0.0.0/8,192.168.0.0/16,172.16.0.0/12'
dockernet: '172.17.0.0'
mdengine: 'ZEEK'
ids: 'Suricata'
url_base: '10.'
managerip: '10.'
```

**Reply by wolfmatt21** [reply]
I think i have found from your message @m0duspwnens. Any clue how i edit that conf?

![image](https://user-images.githubusercontent.com/82594848/136203133-5a4575a6-8ee3-4329-9a24-839dcb6b5d54.png)

**Reply by wolfmatt21** [reply]
Actually can i just remove that sensor from the master, as the conf file isnt in /opt/so/saltstack/local/pillar/minions file?

**Comment by m0duspwnens** [comment, marked as answer]
Check the file located at `/opt/so/saltstack/local/pillar/data/nodestab.sls`, and you can remove the entire entry for that node, or modify the IP if needed. This is what is causing the issue the elasticsearch state not stating the docker container.

If you no longer need that node, then (from the manager )you could remove its minion pillar file from `/opt/so/saltstack/local/pillar/minions/`, remove its salt key with `salt-key -d <minionid>`, then remove its IP from the firewall allow list using `so-firewall`.

Could you fill us in on any details that may have lead to the `2` being inserted as the IP for that node? Edit: Looking at the screen capture, I can see that somehow the data for that node is scattered. Example, the `guid` field has the interface name, the `ip` is likely the `totalcpus`.

Did this issue start after that node was added to the grid or did it begin after a soup? Does that node have multiple IPs assigned to the `enp1s0` interface?

**Reply by wolfmatt21** [reply]
To be honest i had an incline something went wrong during setup of that node but i didnt know where. Thats all fixed now thank you for your help!
